api/src/main/java/module-info.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 International Business Machines Corp. and others
+ * 
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Licensed under the Apache License, 
+ * Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+module jakarta.batch {
+
+    // Used in BatchRuntime
+    requires java.logging;
```

Comment:
I'm not a JPMS expert, but why do we need java.logging for the API? Sure, the runtime (implementation) needs it but the API? Can't we just remove `requires java.logging`? I know that it's not a big deal if it stays, it's a built-in module in Java SE and to remove it one would need to build their own Java runtime without the java.logging module which doesn't make much sense, so I don't care much, just asking.

Reply:
Thanks for your review.

We use logging in the [BatchRuntime.java](https://github.com/eclipse-ee4j/batch-api/blob/d49675492b5f1f77f2d53236e87d6f7ee1bd8f78/api/src/main/java/jakarta/batch/runtime/BatchRuntime.java#L49-L51).  This isn't required by the spec, but it's a detail of the API jar published by oure spec project.)   

Like you said, I'm not sure how you can build a runtime that doesn't have `java.logging` so it seemed OK to me.

Reply:
You can have runtimes without it for sure, using slf4j or log4j2 makes java.logging nt required by the impl so the spec shouldnt mandate it - as mentionned earlier the logging of BatchRuntime can be dropped if it helps, nobody uses it anyway.

Reply:
The spec doesn't mandate it.   Someone is free to build their own API JAR  and use a custom module-info.   (it just needs to pass the SigTest to be spec-compliant...AFAIK there's no validation of module-info).    I think it's more useful to have logging.  I suppose we could switch to `System.Logger` instead though and be able to drop the `java.logging` dependency.

Reply:
Does it mean jbatch wants to encourage to not use its own jar? It is how I read it :s.

Reply:
Hi @scottkurz, your comments suggest that there would be a problem with the impl if we remove the dependency on java.logging, is that right? Does the impl need it in the API? I would think that if the impl needs logging, it would contain its own module-info.java in its impl JAR and define the dependency there instead, am I wrong?

I tend to agree with @rmannibucau if he insists that having the dependency could cause issues in some implementations. I don't know if there is or ever will be any impl that doesn't want to include java logging but if there's even a slight chance, I would support that now rather than revisit that in a later version of the spec after some issues are reported. 

I'd like to see Jakarta EE, including Batch, as a platform that makes expected things simpler and easier to do. We can wait for innovative solutions to emerge and prove themselves before we standardize them but then we should at least try to create the purest and easy to use solution based on our current knowledge. I believe that @rmannibucau backs his suggestion with his own experience and therefore we shouldn't ignore it and try to find a nicer solution than we have today.

Reply:
To clairify: if we intend to support JPMS we should do it or we shouldn't add a module-info at all IMHO.
JBatch does not require java.logging so it shouldn't be there (it is an API internal leak) so let's drop it.
One big drawback of having it is you can impact jimage usage in an way which is unintentional and the final presence of JUL will change the runtime behavior quite commonly with logging facades so don't think it is sane even if it does not kill completely the spec.
Let see it as an opportunity to clean the buggy BatchRuntime impl ;).